src/Avalonia.Diagnostics/Diagnostics/Views/PropertyValueEditorView.cs

```diff
@@ -261,6 +231,74 @@ private static bool ImplementsInterface<TInterface>(Type type)
             }
 
             return tb;
+
+            TControl CreateControl<TControl>(AvaloniaProperty valueProperty
```

Comment:
A few nits here about formatting.

 1. Technically there is a formatting rule some place where either all parameters should be on one line or all parameters on separate lines. Not the first parameter with the method name and the remainder on separate lines. I dont really care about this personally though.
 2. I don't think we should be putting commas at the beginning like this. It isn't done anywhere else in this code base and actually I've never seen it done before. Just like with enums and arrays, etc commas need to be at the end of the line before.

At some point we really need to write formatting rules in a doc for contributos. For the most part people just follow the C# standard convention as it evolves.

Reply:
You are right that there would be a need to write some formatting rules.

I used this style of formatting because it makes it easier to read, you can comment out individual parameters without `/* */`, it makes it easier to identify what changes you've added.

consider the following code fragment:

### current style:

```csharp
public static void MyVeryLongMethod(int a /* a coefficient*/, int b /* b coefficient*/, int c /* c coefficient*/, int d /* d coefficient*/, int e /* e coefficient*/)
{
}
```

### code-style rule like `c`

```csharp
public static void MyVeryLongMethod(int a // a coefficient
             , int b // b coefficient
             , int c // c coefficient
             , int d // d coefficient
             , int e // e coefficient
             )
{

}
```

if add parameter f and change c parameter type

### current style:

```diff
-public static void MyVeryLongMethod(int a /* a coefficient*/, int b /* b coefficient*/, int c /* c coefficient*/, int d /* d coefficient*/, int e /* e coefficient*/)
+public static void MyVeryLongMethod(int a /* a coefficient*/, int b /* b coefficient*/, bool c /* c coefficient*/, int d /* d coefficient*/, int e /* e coefficient*/, string? f = default)
{
}
```

###  code-style rule like `c`

```diff
public static void MyVeryLongMethod(int a // a coefficient
             , int b // b coefficient
-            , int c // c coefficient
+            , bool c // c coefficient
             , int d // d coefficient
             , int e // e coefficient
+            , string? f = default
             )
{

}
```
With a code-style rule like `c`, changes are identified faster.

Reply:
Interesting. Doesn't it work also like this ?

```cs
        void TestMe(
            int a, // Comment 1 goes here
            int b, // comment 2
            int c) // comment 3
        {
            Debug.Print("{a} {b} {c}");
        }
```

Reply:
Yes, but the disadvantage if you add `d` parameter in diff has 3 row changed but only one is significative.

```diff
        void TestMe(
            int a, // Comment 1 goes here
            int b, // comment 2
-           int c, // comment 3
-           int c) // comment 3
+           int d) // comment 4
        {
            Debug.Print("{a} {b} {c}");
        }
```

with code-style rule like c

```csharp
        void TestMe( int a // Comment 1 goes here
            , int b  // comment 2
            , int c // comment 3
            ) 
        {
            Debug.Print("{a} {b} {c}");
        }
```

```diff
        void TestMe( int a // Comment 1 goes here
            , int b  // comment 2
            , int c // comment 3
+           , int d // comment 4
            ) 
        {
            Debug.Print("{a} {b} {c}");
        }
```

Reply:
@workgroupengineering I don't argue that there are good reasons to format the comma as you did. It definitely helps with change tracking on a per-line basis. However, it violates the standard C# convention that goes back decades. I think in this case we need to follow the existing code base.

Reply:
it's not said that since it's always been like this, you have to continue even though there's probably a better way. That said, as soon as I can, I will change.

Reply:
True, and I say that many times myself! In this case though, I don't think the pros outweigh the cons. (Cons being namely its difficult to read at first and changes the convention used everywhere else)

Reply:
Identifying changes more easily, in my opinion is a big pro to improve the quality of the code. That said, I made the changes you suggested.